### PR TITLE
Add a button style for being used on primary like in the header

### LINKED ISSUE #2869: Add Button style for a button on top of primary color
E.g. this could be used in Talk and all other apps that add buttons to the header.
Ref https://github.com/nextcloud/spreed/pull/7593

The problem is the icon+text color would need to toggle based on the primary color instead of background, and so on.

### PR DESCRIPTION
**Pull request by nickvergessen**
Bildschirmfoto vom 2022-07-21 16-14-07

Fix #2869

### CODE REVIEW

**Comment by nickvergessen** [comment]
And yes, the action menu is following the wrong color there as it is not using the vue component yet

**Review by Pytal — APPROVED** [review]

File: src/components/Button/Button.vue [hunk 1]
```diff
@@ -595,6 +600,18 @@ export default {
 		}
 	}
 
+	// Tertiary on primary color (like the header)
+	&--vue-tertiary-on-primary {
```
Comment (marcoambrosini) [inline src/components/Button/Button.vue:604]: Should we just call it inverted colours  or something like that? I can imagine we could use something similar in talk's call view

**Review by marcoambrosini — APPROVED** [review]

File: src/components/Button/Button.vue [hunk 2]
```diff
@@ -535,6 +535,11 @@ export default {
 		&.button-vue--vue-tertiary-no-background {
 			opacity: 1;
 		}
+		&.button-vue--vue-tertiary-on-primary {
+			border-radius: var(--border-radius);
```
[the remark this answers is not in the document]
Comment (nickvergessen) [reply src/components/Button/Button.vue:539]: This is what all the icons in the top bar use:
https://github.com/nextcloud/server/blob/87320cba99229e065747d699461b1dddc54326f8/core/css/header.scss#L24-L28

File: src/components/Button/Button.vue (hunk 1 above)
Comment (nickvergessen) [reply src/components/Button/Button.vue:604]: No that's not how it works. In Talk we would always need white or something alike, not depening on the primary color nor the background.

File: src/components/Button/Button.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcoambrosini) [reply src/components/Button/Button.vue:539]: Yes but we need this kind of buttons elsewhere too

[the remark this answers is not in the document]
Comment (nickvergessen) [reply src/components/Button/Button.vue:539]: I think the square (if we introduce it) should work on all buttons, so I would like to have that done in a seperate PR

**Comment by danxuliu** [comment]
> And yes, the action menu is following the wrong color there as it is not using the vue component yet

For reference: https://github.com/nextcloud/server/issues/33229

Independently of that, I have noticed that in the _:active_ state (which can be triggered by pressing on the button and then moving the cursor away so it no longer hovers it, which overrides _:active_) a round background is shown for the button, even if the box shadow is (almost) square when focused. Is it known / expected? Or should I open an issue?
Tertiary-On-Primary-Active

**Comment by nickvergessen** [comment]
Not expected, please raise an issue

**Pull request merged**